programs/stake/src/stake_state.rs

```diff
@@ -6666,6 +6666,120 @@ mod tests {
         }
     }
 
+    /// Ensure that `split()` correctly handles prefunded destination accounts.  When a destination
+    /// account already has funds, ensure the minimum split amount reduces accordingly.
+    #[test]
+    fn test_split_destination_minimum_stake_delegation() {
+        let rent = Rent::default();
+        let rent_exempt_reserve = rent.minimum_balance(std::mem::size_of::<StakeState>());
+        let source_pubkey = Pubkey::new_unique();
+        let source_meta = Meta {
+            rent_exempt_reserve,
+            ..Meta::auto(&source_pubkey)
+        };
+        // Set the source's starting balance to something large to ensure its post-split
+        // balance meets all the requirements
+        let source_starting_balance = u64::MAX;
+        let source_account = AccountSharedData::new_ref_data_with_space(
+            source_starting_balance,
+            &StakeState::Initialized(source_meta),
```

Comment:
Sorry to be pedantic on this, but any chance we could also have this set of tests for `StakeState::Stake`?  The only other addition there is to check that `split.stake.delegation.stake <= source.stake.delegation.stake`, depending on the rent exempt reserve.

For example, if the destination has its full rent exempt reserve or more, then check `split.stake.delegation.stake == split_lamports`, but if it has less, then check `split.stake.delegation.stake == split_lamports - split_rent_exempt_reserve + dest_starting_balance`

You can also feel free to do this in a separate PR.

Reply:
Phew, took me a bit to grok this, but I'm glad you pointed this out. I've expanded the test with cases for `StakeState::Stake` and also check the destination's stake delegation is expected.

Reply:
Bah sorry about that, I wasn't sure if it was clear -- please feel free to ask for clarification if I'm being too opaque!  Thankfully, you perfectly understood what I meant :grin: